static/js/message_list_view.js

```diff
@@ -740,6 +741,7 @@ export class MessageListView {
         const restore_scroll_position = () => {
             if (
                 !recent_topics_util.is_visible() &&
+                !inbox_util.is_visible() &&
```

Comment:
This is another call point that should definitely be "Is the message list visible".